packages/qwik/src/qwikloader.ts

```diff
@@ -99,7 +99,11 @@ export const qwikLoader = (doc: Document, hasInitialized?: number) => {
                 element: element,
                 reqTime,
               });
-            await handler(ev, element);
+            const results = handler(ev, element);
+            // sync$ may not be async function and e.stopPropagation() won't work unless it's fired immediately
+            if (results && typeof results.then === 'function') {
```

Comment:
I'm trying to avoid `instanceof Promise` here in favor for thenable but maybe it's less code just using instanceof Promise